Code/Tools/AssetProcessor/native/AssetManager/AssetRequestHandler.h

```diff
@@ -249,7 +249,7 @@ namespace AssetProcessor
         // Invokes the appropriate handler and returns true if the message should be deleted by the caller and false if the request handler is responsible for deleting the message
         virtual bool InvokeHandler(MessageData<AzFramework::AssetSystem::BaseAssetProcessorMessage> message);
 
-    private:
+    protected:
```

Comment:
Seems dangerous to make all these methods protected. Do all of them need to have weaker access control? You also have made class members vars protected here, which really breaks encapsulation.

Class also looks strange with two large protected blocks, separated by one public method

Reply:
I've updated this to move just what I needed to protected.